aquifer/aquifer
# Composer support

This PR combines:
- @generalredneck's Composer support work from #131 
- My work from #130 to allow more flexible sync directory configuration.
- Some additional stuff to get `aquifer build` to work using Composer with minimal setup fuss.

**To test:**
- Create a D8 site by running `aquifer create test -d 8 && cd test`
- Edit the `aquifer.json` file changing the `build` object properties to look like so:
  
  ```
  "build": {
    "method": "composer",
    "directory": "build",
    "makeFile": "composer.json"
  }
  ```
- Run `aquifer build`
- Verify that the `/build` directory contains a Drupal site and the synced directories and files defined in `aquifer.json` are synced appropriately.
- Run `mkdir -p config/base`
- Add the following to the `sync.directories` object in `aquifer.json`:

```
"config": {
  "destination": "sites/default/config",
  "method": "symlink",
  "conflict": "overwrite"
}
```
- Run `aquifer build` and verify the new `/config` directory is symlinked into `/build/sites/default`.
- Run `touch root/LICENSE.txt`
- Add the following to the `sync.files` object in `aquifer.json`:

```
"root/LICENSE.txt": {
  "destination": "LICENSE.txt",
  "method": "copy",
  "conflict": "overwrite"
}
```
- Run `aquifer build` and verify `/build/LICENSE.txt` is overwritten by a copy (not symlink) or your blank file.
- In `aquifer.json` change the conflict value to skip for `root/LICENSE.txt`.
- Run `aquifer build` and verify `/build/LICENSE.txt` is has been left alone.
